src/haproxy.py

```diff
@@ -92,9 +94,17 @@ def _render_file(self, path: Path, content: str, mode: int) -> None:
         # Set the correct ownership for the file.
         os.chown(path, uid=u.pw_uid, gid=u.pw_gid)
 
-    def _render_haproxy_config(self) -> None:
-        """Render the haproxy configuration file."""
+    def _render_haproxy_config(self, config: CharmConfig) -> None:
+        """Render the haproxy configuration file.
+
+        Args:
+            config: charm config
+        """
         with open("templates/haproxy.cfg.j2", "r", encoding="utf-8") as file:
             template = Template(file.read(), keep_trailing_newline=True)
-        rendered = template.render()
+        rendered = template.render(config_global_max_connection=config.global_max_connection)
         self._render_file(HAPROXY_CONFIG, rendered, 0o644)
+
+    def _reload_haproxy_service(self) -> None:
+        """Restart the haporxy service."""
+        systemd.service_reload(HAPROXY_SERVICE)
```

Comment:
Perhaps do some error handling and use the Reload error here ?
Also the docstring should be changed to "reload" and there's a typo in "haproxy"